src/Microsoft.AspNetCore.Sockets.Client/Connection.cs

```diff
@@ -166,18 +173,36 @@ public Task StartAsync(ITransport transport, HttpClient httpClient)
         {
             try
             {
-                _logger.LogTrace("Beginning receive loop");
+                _logger.LogTrace("Beginning receive loop.");
 
                 while (await Input.WaitToReadAsync())
                 {
+                    if (_connectionState != ConnectionState.Connected)
```

Comment:
Yes, I just want to call out that's this "stops" it from happening. In quotes because it can still happen. We didn't fix the race, just reduced the annoyance of it.